SoObjects/Mailer/SOGoMailBaseObject.m

```diff
@@ -206,6 +220,22 @@ - (NGImap4Connection *) imap4Connection
       [imap4 retain];
     }
 
+   // Connection broken, try to reconnect
+   if (![[[imap4 client] isConnected] boolValue])
```

Comment:
Please @sixstone-qq check again, I think new code tackles with that situation, from my tests now it works both for connection creation and reconnect